File: dart/glue/lib/src/lib/bool/is_exist.dart
import 'package:glue/src/eval.dart';
import 'package:glue/src/ir.dart';

final Ir isExist_ = IrNativeFunc(isExistImpl);

Eval<Ir> isExistImpl(Ir arg) => Eval.pure(IrBool(isExist(arg)));

File: haskell/glue/src/Glue/Lib/Bool/IsExist.hs
module Glue.Lib.Bool.IsExist where

import Glue.Eval (Eval)
import Glue.IR (IR (..), isExist)

isExist_ :: IR Eval
isExist_ = NativeFunc isExistImpl

isExistImpl :: IR Eval -> Eval (IR Eval)
isExistImpl = pure . Bool . isExist
